tests/auth_tests/test_templates.py

```diff
@@ -2,53 +2,70 @@
 from django.contrib.auth.models import User
 from django.contrib.auth.tokens import PasswordResetTokenGenerator
 from django.contrib.auth.views import (
-    PasswordChangeDoneView, PasswordChangeView, PasswordResetCompleteView,
-    PasswordResetConfirmView, PasswordResetDoneView, PasswordResetView,
+    INTERNAL_RESET_SESSION_TOKEN, PasswordChangeDoneView, PasswordChangeView,
+    PasswordResetCompleteView, PasswordResetDoneView, PasswordResetView,
 )
 from django.test import RequestFactory, TestCase, override_settings
+from django.urls import reverse
 from django.utils.encoding import force_bytes, force_text
 from django.utils.http import urlsafe_base64_encode
 
+from .client import PasswordResetConfirmClient
+
 
 @override_settings(ROOT_URLCONF='auth_tests.urls')
 class AuthTemplateTests(TestCase):
 
-    def test_titles(self):
+    def setUp(self):
         rf = RequestFactory()
         user = User.objects.create_user('jsmith', 'jsmith@example.com', 'pass')
         user = authenticate(username=user.username, password='pass')
         request = rf.get('/somepath/')
         request.user = user
+        self.user, self.request = user, request
 
-        response = PasswordResetView.as_view(success_url='dummy/')(request)
+    def test_PasswordResetView(self):
+        response = PasswordResetView.as_view(success_url='dummy/')(self.request)
         self.assertContains(response, '<title>Password reset</title>')
         self.assertContains(response, '<h1>Password reset</h1>')
 
-        response = PasswordResetDoneView.as_view()(request)
+    def test_PasswordResetDoneView(self):
+        response = PasswordResetDoneView.as_view()(self.request)
         self.assertContains(response, '<title>Password reset sent</title>')
         self.assertContains(response, '<h1>Password reset sent</h1>')
 
+    def test_PasswordResetConfirmView_invalid_token(self):
         # PasswordResetConfirmView invalid token
-        response = PasswordResetConfirmView.as_view(success_url='dummy/')(request, uidb64='Bad', token='Bad')
+        client = PasswordResetConfirmClient()
+        url = reverse('password_reset_confirm', kwargs={'uidb64': 'Bad', 'token': 'Bad-Token'})
+        response = client.get(url)
         self.assertContains(response, '<title>Password reset unsuccessful</title>')
         self.assertContains(response, '<h1>Password reset unsuccessful</h1>')
+        self.assertNotIn(INTERNAL_RESET_SESSION_TOKEN, client.session)
```

Comment:
I guess it was to check that `self.request.session.pop(INTERNAL_RESET_SESSION_TOKEN, None)` was correctly removing the token added in the session. As the logic changed a bit, I think this assertion is no more relevant.